src/runtime.rs

```diff
@@ -105,13 +105,35 @@ impl RunState {
     }
 
     #[inline]
-    fn reg(&mut self, reg: u16) -> &mut u16 {
+    fn reg(&self, reg: u16) -> u16 {
+        let value = self.reg.get(reg as usize);
```

Comment:
why not `get_unchecked`? not sure if the debug assert is necessary if every instance of the function being called is trackable and verifiable to be within acceptable bounds

Reply:
I suppose anyone using these methods (including a trap implementor) should be responsible for safety... I used `get` only for the `debug_assert`, but I realize now that it could be written as `debug_assert(reg < 8)` in order to use `get_unchecked` -- if done like this I can't see a reason to omit the assertion.

Reply:
Sounds good, best of both worlds :+1: